### PR TITLE
159 revisit bignumber usage

### PR DESCRIPTION
**Pull request by ibjc**

### CODE REVIEW

**Comment by ibjc** [comment]
@LaneTwo fyi

File: src/components/dashboard/burn_tokens.tsx [hunk 1]
```diff
@@ -310,58 +294,41 @@ export default function BurnTokens(props: mintProps) {
...
+		const supplyDeltaBig = bignumber(inverseTokenSupply.sub(burnedAmount).toString()).dividedBy(bignumber(inverseTokenSupply.toString()))
+		const liquidityReceivedBig = reserveAmount.sub(mulPercent(reserveAmount, supplyDeltaBig.sqrt().toNumber()))
 
 		// calculate spot price post mint
-		const curveInvariant = Number(formatUnits(reserveAmount, defaultDecimals)) / Math.pow(Number(formatUnits(inverseTokenSupply, inverseTokenDecimals)), curveUtilization) 
-		const newPrice = curveInvariant * curveUtilization
-		/
-		Math.pow(Number(formatUnits(inverseTokenSupply.sub(burnedAmount), inverseTokenDecimals)), 1 - curveUtilization)
+		const curveInvariantBig = BigNumber.from(bondingCurveParams.invariant)
+		const newPriceBig = curveInvariantBig.mul(parseUnits(curveUtilization.toString(), defaultDecimals)).div(computeSquareRoot(inverseTokenSupply.sub(burnedAmount))).div(parseUnits("1", defaultDecimals/2))
```
Comment (LaneTwo) [inline src/components/dashboard/burn_tokens.tsx:320]: From contract side, price is calculated by: utilization * reserve / ibcSupply

**Review by LaneTwo — APPROVED** [review]

File: src/components/dashboard/burn_tokens.tsx [hunk 2]
```diff
@@ -372,14 +339,18 @@ export default function BurnTokens(props: mintProps) {
...
-		const liquidityReceived = parsedAmount / (1 - totalFeePercent) // full mint-amount
-
-		setLiquidityReceived(parseUnits(liquidityReceived.toFixed(defaultDecimals), defaultDecimals)) 
+		const liquidityReceivedBig = mulPercent(parseUnitsBnJs(parsedAmount, defaultDecimals), 1/(1-totalFeePercent))
+		setLiquidityReceived(liquidityReceivedBig) 
 
 		// calculate ibc burn amount
+		const currentInverseTokenSupplyBig = BigNumber.from(bondingCurveParams.inverseTokenSupply)
 		const currentInverseTokenSupply = Number(ethers.utils.formatUnits(bondingCurveParams.inverseTokenSupply, inverseTokenDecimals.toString()))
 		const k = 1 - curveUtilization
```
Comment (LaneTwo) [inline src/components/dashboard/burn_tokens.tsx:348]: also, contract is using invariant and utilization for calculation rather than m & k now

**Review by LaneTwo — APPROVED** [review]

File: src/components/dashboard/burn_tokens.test.tsx [hunk 3]
```diff
@@ -0,0 +1,119 @@
...
+		await waitFor(() => {
+			expect(newPrice).toBe('1571889231955141735')
+			expect(newIbcIssuance.toString()).toBe('3205453629956167694')
+			expect(newReserve).toBe(ethers.BigNumber.from('21131781542148949646').sub(ethers.BigNumber.from('11054545454545455629')).toString())
+		});
+	});
+});
+
+/*
+2814276007789739754535245660
```
Comment (kimchicoffee) [inline src/components/dashboard/burn_tokens.test.tsx:107]: what's the purpose of this commented out numbers? 0_0??

File: src/components/dashboard/burn_tokens.tsx (hunk 1 above)
Comment (ibjc) [reply src/components/dashboard/burn_tokens.tsx:320]: ah thanks, this is way simpler than the formula on the docs

File: src/components/dashboard/burn_tokens.tsx (hunk 2 above)
Comment (ibjc) [reply src/components/dashboard/burn_tokens.tsx:348]: yea for the mint case, we use the invariant/utilization formula like the contract

for burns, the contract does not have an "exactAmountOut" calculation, and just went with the m&k version; imo the formula complexity is about the same

File: src/components/dashboard/burn_tokens.tsx (hunk 1 above)
Comment (ibjc) [reply src/components/dashboard/burn_tokens.tsx]: fixed in https://github.com/bear-market-labs/ibc_interface/pull/252/commits/655f61def832c9e947bb5e1de05afe131bb8268f

File: src/components/dashboard/burn_tokens.test.tsx (hunk 3 above)
Comment (ibjc) [reply src/components/dashboard/burn_tokens.test.tsx]: fixed in https://github.com/bear-market-labs/ibc_interface/pull/252/commits/2ae49c487912629ec7451edee1619e03bf903aee

**Pull request merged**